storage/innobase/handler/ha_innodb.cc

```diff
@@ -21697,7 +21721,9 @@ static void innodb_encryption_threads_update(
     const void *save) /*!< in: immediate result
                       from check function */
 {
+  mysql_mutex_unlock(&LOCK_global_system_variables);
```

Comment:
I wonder whether this is safe. For example, if two threads issue SET for this var to two different values, will there be a race?